### PR TITLE
build: cleanup old cmake code

### PR DESCRIPTION
**Pull request by dundargoc**
- build: remove ARGS from add_custom_command

It's a command that doesn't do anything, kept only for compatibility
reasons.

- build: remove unnecessary policy related code

Having cmake version 3.10 as the required minimum version ensures these
are set to new by default.

- build: replace deprecated CMAKE_COMPILER_IS_GNUCC variable

Instead use the recommended form `CMAKE_C_COMPILER_ID MATCHES "GNU"`

- build: remove unnecessary build functions

These functions serve no purpose if they're only used as intermediary
functions that passes on arguments to ExternalProject_Add.

### CODE REVIEW

File: cmake.deps/cmake/BuildLibvterm.cmake [hunk 1]
```diff
@@ -66,9 +33,22 @@ else()
                                            install)
 endif()
 
-BuildLibvterm(PATCH_COMMAND ${LIBVTERM_PATCH_COMMAND}
-  CONFIGURE_COMMAND ${LIBVTERM_CONFIGURE_COMMAND}
-  BUILD_COMMAND ${LIBVTERM_BUILD_COMMAND}
-  INSTALL_COMMAND ${LIBVTERM_INSTALL_COMMAND})
+ExternalProject_Add(libvterm
```
Comment (justinmk) [inline cmake.deps/cmake/BuildLibvterm.cmake:36]: Nice!

**Review by justinmk — APPROVED** [review]

**Comment by justinmk** [comment]
still in draft intentionally?

**Comment by dundargoc** [comment]
Yeah, there are two more Build functions I can potentially remove. Let me check them and then I can undraft.

**Comment by dundargoc** [comment]
@justinmk let's call it a day, the PR is ready. There's a lot more code that can be removed but let's not change everything at once.

**Comment by clason** [comment]
Thanks, keep 'em coming!

**Pull request merged**